ouroboros-network-api/src/Ouroboros/Network/AnchoredFragment.hs

```diff
@@ -162,7 +162,7 @@ anchorToPoint (Anchor s h _b) = BlockPoint s h
 -- In this case, we must also be given the 'BlockNo'. This only makes sense
 -- for points that aren't genesis.
 anchorFromPoint :: Point block -> BlockNo -> Anchor block
-anchorFromPoint GenesisPoint _     = error "anchorFromPoint: genesis point"
+anchorFromPoint GenesisPoint _     = error "anchorFromPoint: genesis point" -- TODO
```

Comment:
Two points here:
- It seems, to my untrained eye, like the natural result of this function might be `AnchorGenesis`, rather than an error.
- This function appears unused across your org's repos. It may be a candidate for deprecation and/or deletion.

Thoughts?

Reply:
The final sentence of the Haddock explains, I think.

> `This only makes sense for points that aren't genesis`

Shot in the dark: It may be the case, then, that the first argument should instead be a `RealPoint`?

Reply:
I am not sure if it makes sense to not allow anchors from `Genesis`, in principle it could make things more uniform (one doesn't need to treat `Genesis` in a special way), but maybe it makes sense from `consensus` perspective.

I don't know what `RealPoint` is, maybe it's something in `ouroboros-consensus`.

Reply:
My apologies. It's the last _two_ sentences, not just the last sentence.

> In this case, we must also be given the 'BlockNo'. This only makes sense for points that aren't genesis.

The caller of this function is providing a `Point` and a `BlockNo` that refer to the same block. Notably, the genesis "block" does not have a block no; block no 0 identifies the first _real_ block, ie the successor of the genesis block. Hence, the existence of the given `BlockNo` implies that the given `Point` is _not_ `GenesisPoint`.

The quotient type "`Point` but not Genesis" is isomorphic to what Consensus defines to be the `RealPoint` type.

Reply:
I see - thanks for the clarification, @nfrisby. I can imagine a few paths forward here, ordered roughly by how much turbulence I think they would entail:
1. Leave everything alone.
2. Make this function explicitly partial - `Point block -> Maybe BlockNo -> Maybe (Anchor block)`, for instance, could reflect a semantics that respects these genesis/`BlockNo` concerns.
3. Deprecate and delete the function.
4. Make the argument a `RealPoint` - which, I think, would entail moving `RealPoint` from `ouroboros-consensus` to `ouroboros-network-api`.

Do y'all have thoughts or preferences among these options?

Reply:
You might need to ping @nfrisby on slack.

Reply:
> I am not sure if it makes sense to not allow anchors from Genesis

Are you suggesting removing the `AnchorGenesis` constructor? I anticipate that that one _is_ used in the codebase.

---

My intuition favors options 3 or 4. If it's unused, removing it as dead-code seems wise. On the other hand, it's a very reasonable smart constructor to expect for the data type---in that case, I would favor renaming it to `anchorFromRealPoint`, eg.

---

I would I don't know if `RealPoint` would be useful in the `ouroboros-network` codebase---@marcin?

https://github.com/input-output-hk/ouroboros-consensus/blob/71e6cbb304d89ba7abe32c8d84239014ca24c94e/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/RealPoint.hs#L41C1-L43

Reply:
To be clear, my suggestion that `RealPoint` be moved derives from the dependency of `ouroboros-consensus` on `ouroboros-network-api`, though I imagine it could also be defined in another package entirely. At any rate, I've opened #4690 to continue this discussion, thinking that conversations on moving structures between packages or deprecation of functions warrant more visibility.

Reply:
> I would I don't know if RealPoint itself would otherwise be useful in the ouroboros-network codebase---@marcin?

Other than the use case here, we don't need `RealPoint`, but I'd be fine with moving it to  `Ouroboros.Network.Block`.